File: src/lib/php/DSDatabase.php
<?php

require_once "DSLib.php";
require_once "DSClassInfo.php";

define("DS_DATABASE_VERSION", "0.0.1");
define("DS_DATABASE_NAME",    "DSTemplate");
define("DS_DATABASE_AUTHOR",  "Thorsten Geppert (thorsten@ext2fs.de)");

abstract class DSDatabase extends DSLib {

	abstract public function Connect();
	abstract public function IsConnected();
	abstract public function Reconnect();
	abstract public function Disconnect();

	protected $hostname = "";
	protected $database = "";
	protected $username = "";
	protected $password = "";
	protected $port     = "";

	public function GetClassInfo() {
		return new DSClassInfo(
			DS_DATABASE_NAME,
			DS_DATABASE_VERSION,
			DS_DATABASE_AUTHOR
		);
	}

	public function __construct(
		$hostname,
		$database,
		$username,
		$password,
		$port
	) {
		parent::__construct();

		$this->SetHostname($hostname);
		$this->SetDatabase($database);
		$this->SetUsername($username);
		$this->SetPassword($password);
		$this->SetPort($port);
	}

	public function SetHostname($hostname) {
		$this->hostname = $hostname;
	}

	public function GetHostname() {
		return $this->hostname;
	}

	public function SetDatabase($database) {
		$this->database = $database;
	}

	public function GetDatabase() {
		return $this->database;
	}

	public function SetUsername($username) {
		$this->username = $username;
	}

	public function GetUsername() {
		return $this->username;
	}

	public function SetPassword($password) {
		$this->password = $password;
	}

	public function GetPassword() {
		return $this->password;
	}

	public function SetPort($port) {
		$this->port = $port;
	}

	public function GetPort() {
		return $this->port;
	}

}

?>

File: src/lib/cpp/DSDatabase.cpp
#include "DSDatabase.h"

DynSoft::DSDatabase::DSDatabase() : DSLib() {
}

DynSoft::DSDatabase::DSDatabase(
	const wxString &filename,
	const wxString &username,
	const wxString &password
) : DSLib() {
	this->filename = filename;
	this->username = username;
	this->password = password;
}

DynSoft::DSDatabase::DSDatabase(
	const wxString &hostname,
	const wxString &database,
	const wxString &username,
	const wxString &password,
	const wxString &port
) : DSLib() {
	this->hostname = hostname;
	this->database = database;
	this->username = username;
	this->password = password;
	this->port     = port;
}

DynSoft::DSDatabase::~DSDatabase() {
}

DynSoft::DSClassInfo DynSoft::DSDatabase::GetClassInfo() {
	return DynSoft::DSClassInfo(
		DS_DATABASE_NAME,
		DS_DATABASE_VERSION,
		DS_DATABASE_AUTHOR
	);
}

void DynSoft::DSDatabase::SetFilename(const wxString &filename) {
	this->filename = filename;
}

wxString DynSoft::DSDatabase::GetFilename() const {
	return filename;
}

void DynSoft::DSDatabase::SetHostname(const wxString &hostname) {
	this->hostname = hostname;
}

wxString DynSoft::DSDatabase::GetHostname() const {
	return hostname;
}

void DynSoft::DSDatabase::SetDatabase(const wxString &database) {
	this->database = database;
}

wxString DynSoft::DSDatabase::GetDatabase() const {
	return database;
}

void DynSoft::DSDatabase::SetUsername(const wxString &username) {
	this->username = username;
}

wxString DynSoft::DSDatabase::GetUsername() const {
	return username;
}

void DynSoft::DSDatabase::SetPassword(const wxString &password) {
	this->password = password;
}

wxString DynSoft::DSDatabase::GetPassword() const {
	return password;
}

void DynSoft::DSDatabase::SetPort(const wxString &port) {
	this->port = port;
}

wxString DynSoft::DSDatabase::GetPort() const {
	return port;
}

